addons/purchase/purchase.py

```diff
@@ -149,6 +149,20 @@ def _get_journal(self, cr, uid, context=None):
                                                 limit=1)
         return res and res[0] or False  
 
+    def _has_non_stockable_item(self, cr, uid, ids, *args):
+        res = dict.fromkeys(ids, False)
+        for order in self.browse(cr, uid, ids):
+            for order_line in order.order_line:
+                if (
+                    not order_line.product_id
+                    or
+                    order_line.product_id
+                    and order_line.product_id.type not in ('product',
+                                                           'consu')
+                ):
+                    res[order.id] = True
+        return res
```

Comment:
I don't agree.  It should be if not(all([x.product_id.type not in ('product', 'consu') for x in order.order_line]), so has_only_non_stockable_item